lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -158,7 +159,52 @@ sub cache_assets {
         symlink($asset, basename($asset)) or die "cannot create link: $asset, $pooldir";
         $vars->{$this_asset} = path(getcwd, basename($asset))->to_string;
     }
-    return;
+    return undef;
+}
+
+sub sync_tests {
+    my ($shared_cache) = @_;
+    my $cache_client   = OpenQA::Worker::Cache::Client->new;
+    my $rsync_request  = $cache_client->request->rsync(
+        from => $hosts->{$current_host}{testpoolserver},
+        to   => $shared_cache
+    );
+
+    # TODO: Enqueue all requests in one place
+    return {error => "Failed to send rsync cache request"} unless $rsync_request->enqueue;
+
+    sleep 5 and update_setup_status until $rsync_request->processed;
+    # treat "no sync necessary" as success as well
+    my $exit = $rsync_request->result // 0;
+
+    if (my $output = $rsync_request->output) { log_info("rsync: " . $output, channels => 'autoinst') }
+    return $exit;
+}
+
+# do asset caching if CACHEDIRECTORY is set
+sub do_asset_caching {
+    my ($vars) = @_;
+    my $shared_cache;
+    my $assetkeys = detect_asset_keys($vars);
+    if (!$worker_settings->{CACHEDIRECTORY}) {
+        my $error = locate_local_assets($vars, $assetkeys);
+        return $error if $error;
+        return undef;
+    }
+
+    my $host_to_cache = OpenQA::Worker::Cache::_base_host($current_host);
+    my $error         = cache_assets($job => $vars => $assetkeys);
+    return $error if $error;
+
+    # do test caching if TESTPOOLSERVER is set
+    if ($hosts->{$current_host}{testpoolserver}) {
+        $shared_cache = catdir($worker_settings->{CACHEDIRECTORY}, $host_to_cache);
+        $vars->{PRJDIR} = $shared_cache;
+        my $exit = sync_tests($shared_cache);
+        return {error => "Failed to rsync tests: exit code: $exit"} unless (defined $exit && $exit == 0);
+        $shared_cache = catdir($shared_cache, 'tests');
+    }
+    return $shared_cache;
 }
```

Comment:
Putting test functions here will make refactoring much harder.

Reply:
hm, wait something seems to have gone wrong. It was not my intention to add any test code here of course. I will have to check what I was doing.

Reply:
fixed, was a rebase gone wrong. I have extracted both methods "do_asset_caching" and "sync_tests" properly now and covered by according unit tests.